### PR TITLE
Add unit test for components

### PR DESCRIPTION
**Pull request by nguyenson2012**
Add unit test for AlertContainer and Button components

### CODE REVIEW

File: __tests__/components/AlertContainer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  });
+
+  it('renders successfully with its title', async () => {
+    const { baseElement, getByText } = render(<AlertContainer {...props} />);
+
+    expect(baseElement).toMatchSnapshot();
+    expect(getByText(alertTilte)).toBeDefined();
+  });
+
+  it('renders successfully with its message', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/AlertContainer.test.tsx]: merge into 1 test with previous test

File: __tests__/components/AlertContainer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    jest.resetAllMocks();
+  });
+
+  it('renders successfully', async () => {
+    const { baseElement } = render(<AlertContainer {...props} />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders successfully with its title', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/AlertContainer.test.tsx]: merge into 1 test with previous test

File: __tests__/components/AlertContainer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  it('handle press when click default close button', async () => {
+    const { getByText } = render(<AlertContainer {...props} />);
+    const closeButton = getByText(closeText);
+
+    fireEvent.press(closeButton);
+
+    expect(closeAction).toBeCalledTimes(1);
+  });
+
+  it('handle press when clicking custom button', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/AlertContainer.test.tsx]: update type 'handle' to 'handles'

File: __tests__/components/AlertContainer.test.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    const { getByText } = render(<AlertContainer {...propWithActions} />);
+    const cancelButton = getByText(cancelText);
+    const customButton = getByText(customButtonText);
+
+    fireEvent.press(cancelButton);
```
Comment (thinhtran3588) [inline __tests__/components/AlertContainer.test.tsx]: should only test 1 click event

File: __tests__/components/Card.test.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+describe('components/Card', () => {
+  const helloText = 'Hello';
+  const buttonText = 'button';
+
+  beforeEach(() => {
+    jest.spyOn(useTheme, 'useTheme').mockImplementation(() => ({
+      changePrimaryColor: () => {},
+      changeTheme: () => {},
+      componentBackgroundColor: 'grey',
```
Comment (thinhtran3588) [inline __tests__/components/Card.test.tsx]: should make value the same as variable name

File: __tests__/components/AlertContainer.test.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+  it('renders successfully with its message', async () => {
+    const { baseElement, getByText } = render(<AlertContainer {...props} />);
+
+    expect(baseElement).toMatchSnapshot();
+    expect(getByText(message)).toBeDefined();
+  });
+
+  it('renders successfully with its default close button', async () => {
+    const { baseElement, getByText } = render(<AlertContainer {...props} />);
```
Comment (thinhtq-gosei) [inline __tests__/components/AlertContainer.test.tsx]: merge into 1 test with previous test

File: __tests__/components/Button.test.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  });
+
+  it('renders undefined children successfully', async () => {
+    const { baseElement } = render(<Button {...props} width={100} />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders children successfully', async () => {
+    const { baseElement, getByText } = render(
```
Comment (thinhtran3588) [inline __tests__/components/Button.test.tsx]: merge into the default test case

File: __tests__/components/Card.test.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders with primary color successfully', async () => {
+    const { baseElement } = render(<Card primary />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders children successfully', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/Card.test.tsx]: merge intot the default test case

File: __tests__/components/Card.test.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      <Card>
+        <Text>{helloText}</Text>
+      </Card>,
+    );
+
+    expect(baseElement).toMatchSnapshot();
+    expect(getByText(helloText)).toBeDefined();
+  });
+
+  it('renders successfully with Text and Button', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/Card.test.tsx]: merge intot the default test case

File: __tests__/components/CardItem.test.tsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  });
+
+  it('renders with primary color successfully', async () => {
+    const { baseElement } = render(<CardItem primary />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders with theme light color successfully', async () => {
+    jest.spyOn(useTheme, 'useTheme').mockImplementation(() => ({
```
Comment (thinhtran3588) [inline __tests__/components/CardItem.test.tsx]: refactor code to use mockTheme

File: __tests__/components/CardItem.test.tsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  });
+
+  it('renders with bordered card item successfully', async () => {
+    const { baseElement } = render(<CardItem bordered header={false} />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders children successfully', async () => {
+    const { baseElement, getByText } = render(
```
Comment (thinhtran3588) [inline __tests__/components/CardItem.test.tsx]: merge into the default test case

File: __tests__/components/CardItem.test.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+      <CardItem>
+        <Text>{helloText}</Text>
+      </CardItem>,
+    );
+
+    expect(baseElement).toMatchSnapshot();
+    expect(getByText(helloText)).toBeDefined();
+  });
+
+  it('renders successfully with Text and Button', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/CardItem.test.tsx]: merge intot the default test case

File: __tests__/components/DatePicker.test.tsx [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+        primaryColorCode: 'red',
+        theme: 'theme',
+      },
+    }),
+  },
+}));
+
+describe('components/DatePicker', () => {
+  beforeEach(() => {
+    jest.spyOn(useTheme, 'useTheme').mockImplementation(() => ({
```
Comment (thinhtran3588) [inline __tests__/components/DatePicker.test.tsx]: use mockTheme

File: __tests__/components/ErrorText.test.tsx [hunk 14]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { ErrorText } from '@app/components';
+import { mockTheme } from '../helper/mock_theme';
+
+beforeAll(() => {});
```
Comment (thinhtran3588) [inline __tests__/components/ErrorText.test.tsx]: remove

File: __tests__/components/ErrorText.test.tsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    mockTheme();
+  });
+
+  it('renders successfully', async () => {
+    const { baseElement } = render(<ErrorText />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('renders successfully with content', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/ErrorText.test.tsx]: merge into the default test case

File: __tests__/components/Field.test.tsx [hunk 16]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import { Field, Picker, PickerDataItem } from '@app/components';
+import { mockTheme } from '../helper/mock_theme';
+
+beforeAll(() => {});
```
Comment (thinhtran3588) [inline __tests__/components/Field.test.tsx]: remove

File: __tests__/components/Field.test.tsx [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+  const errorText = 'error';
+  const tooltipText = 'tooltip';
+  const testText = 'test';
+  beforeEach(() => {
+    mockTheme();
+  });
+
+  it('renders successfully', async () => {
+    const { baseElement } = render(<Field label={labelText} value={valueText} />);
+    expect(baseElement).toMatchSnapshot();
```
Comment (thinhtran3588) [inline __tests__/components/Field.test.tsx]: add break line

File: __tests__/components/Field.test.tsx [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    const { getByDisplayValue, baseElement } = render(
+      <Field label={labelText} value={valueText} type={'text'} onChangeText={onChangeText} />,
+    );
+
+    expect(baseElement).toMatchSnapshot();
+    fireEvent.changeText(getByDisplayValue(valueText), testText);
+    expect(onChangeText).toBeCalledTimes(1);
+  });
+
+  it('renders picker field successfully', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/Field.test.tsx]: change description to 'shows picker when pressing text (Picker Field)'

File: __tests__/components/List.test.tsx [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+        value: valueItem,
+        primaryColor: colors.blue,
+      },
+    ];
+    const { baseElement } = render(<List data={data} />);
+
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('call function when click successfully', async () => {
```
Comment (thinhtran3588) [inline __tests__/components/List.test.tsx]: fix type call to calls

Comment (thinhtran3588) [reply __tests__/components/List.test.tsx]: click => pressing

File: __tests__/components/Picker.test.tsx [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+  },
+}));
+
+describe('components/Picker', () => {
+  beforeEach(() => {
+    mockTheme();
+  });
+
+  it('show picker successfully', async () => {
+    RNPicker.init = jest.fn();
```
Comment (thinhtran3588) [inline __tests__/components/Picker.test.tsx]: remove unused code

File: __tests__/components/Tooltip.test.tsx [hunk 21]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { Tooltip } from '@app/components';
+import { colors } from '@app/core';
+import { mockTheme } from '../helper/mock_theme';
+
+beforeAll(() => {});
```
Comment (thinhtran3588) [inline __tests__/components/Tooltip.test.tsx]: remove

**Pull request merged**